File: src/lib/commands/git-sync.ts
import { invoke } from "@tauri-apps/api/core";

interface GitSyncEvent {
  status: unknown;
  last_commit_date: number | null;
}

export async function getGitSyncStatus(): Promise<GitSyncEvent> {
  return invoke<GitSyncEvent>("get_git_sync_status");
}

export async function manualSync(): Promise<void> {
  return invoke("manual_sync");
}

export async function setGitSyncEnabled(enabled: boolean): Promise<void> {
  return invoke("set_git_sync_enabled", { enabled });
}

File: src-tauri/src/commands/git_sync.rs
use crate::models::git_status::GitSyncEvent;
use crate::state::git_sync_manager::GitSyncManager;
use crate::state::settings_manager::SettingsManager;
use std::sync::Arc;
use tauri::{AppHandle, State};

#[tauri::command]
pub fn get_git_sync_status(
    git_sync_manager: State<'_, Arc<GitSyncManager>>,
) -> GitSyncEvent {
    git_sync_manager.get_status_event()
}

#[tauri::command]
pub fn manual_sync(
    git_sync_manager: State<'_, Arc<GitSyncManager>>,
    app_handle: AppHandle,
) {
    git_sync_manager.manual_sync(&app_handle);
}

#[tauri::command]
pub fn set_git_sync_enabled(
    enabled: bool,
    git_sync_manager: State<'_, Arc<GitSyncManager>>,
    settings_manager: State<'_, SettingsManager>,
    app_handle: AppHandle,
) {
    settings_manager.update(serde_json::json!({ "git_sync_enabled": enabled }));
    git_sync_manager.set_enabled(enabled, &app_handle);
}
